mutations/loadLock.js

```diff
@@ -70,117 +113,189 @@ async function loadLock(inputs, models, req) {
         }
-    
-        if(LockSearch.OriginalLockType_ID != null) {
-            const OriginalLockType = await loadOriginalLockType(LockSearch);
-            
-            const LoadedLock = await models.LoadedLock.create({
-                CreatedLock_ID: LockSearch.Lock_ID,
-                Lockee: req.Authenticated,
-                Keyholder: LockSearch.User_ID,
-                Code: await NewCode(req.Authenticated) || inputs.Code,
-                Original_Lock_Deck: OriginalLockType.Original_Loaded_ID,
-                Emergency_Keys_Enabled: inputs.Emergency_Keys,
-                Emergency_Keys_Amount: inputs.Emergency_Keys_Amount,
-                Test_Lock: inputs.Test_Lock,
-                Unlocked: false,
-                Free_Unlock: false,
-                Fake_Lock: false
-            })
-            
-            const NumOfFakes = RandomInt(inputs.Min_Fakes, inputs.Max_Fakes);
-
-            if(NumOfFakes > 0) {
-                for(let i = 0; i < NumOfFakes; i++) {
-                    const OriginalLockType = await loadOriginalLockType(LockSearch);
-
-                    await models.LoadedLock.create({
-                        CreatedLock_ID: LockSearch.Lock_ID,
-                        Lockee: req.Authenticated,
-                        Keyholder: LockSearch.User_ID,
-                        Code: await NewCode(req.Authenticated),
-                        Original_Lock_Deck: OriginalLockType.Original_Loaded_ID,
-                        Emergency_Keys_Enabled: inputs.Emergency_Keys,
-                        Emergency_Keys_Amount: inputs.Emergency_Keys_Amount,
-                        Test_Lock: inputs.Test_Lock,
-                        Unlocked: false,
-                        Free_Unlock: false,
-                        Fake_Lock: true,
-                        Real_Lock: LoadedLock.LoadedLock_ID
-                    })
-                }
-            }
-            
-            return LoadedLock;
+    }
+    // tested
+
+    if(validationErrors.length) {
+        throw new UserInputError("Cannot load lock", {
+            invalidArgs: validationErrors
+            });
+    }
+
+    const resultArray = []; 
+
+    /**
+     * realLock is the LoadedLock object that is the real lock.
+     * @type {LoadedLock} */
+    const realLock = await createLoadedLock(LockSearch, req.Authenticated, inputs, true);
+    resultArray.push(realLock);
+
+    if (minFakes > 0) { // maxFakes will be >= minFakes if exception not thrown in validation
+        const NumOfFakes = await RandomInt(minFakes, maxFakes);
+        for(let i = 0; i < NumOfFakes; i++) {
+            /**
+             * fakeLock is a LoadedLock object that is a fake lock.
+             * @type {LoadedLock} */
+            const fakeLock = await createLoadedLock(LockSearch, req.Authenticated, inputs, false, realLock.LoadedLock_ID);
+            resultArray.push(fakeLock);
         }
-    } 
+    } // tested   
+    return resultArray
```

Comment:
we need to shuffle this array here, otherwise the real lock will always be the first one in the list, and as such it will be clear which one is the real one.

Reply:
Also need to shuffle in the query to get current loaded locks, since in the database it will be in the same order

Reply:
True. We'll also need to make sure eventually that this doesn't actually return an array of LoadedLocks where the results would automatically include information the lockee shouldn't see, such as the code, the real_lock link, etc.  I think the way to do this is to add a LockeeLockView to the schema along with a resolver that will only include the appropriate fields.